lib/period.js

```diff
@@ -104,6 +119,44 @@ export default class Period {
     return periodRoot;
   }
 
+  prevPeriodProof() {
+    if (!this.usePrev) {
+      throw Error('not set to use prev period in proofs');
+    }
+
+    const { casRoot,  metaRoot } = this.periodData();
+
+    const proof = [];
+
+    // fees hash
+    proof.push(bufferToHex(Buffer.alloc(32, 0)));
+    proof.push(this.prevHash);
+    proof.push(this.merkleRoot());
+    proof.push(bufferToHex(casRoot));
+
+    return proof;
+  }
+
+  // returns current period
+  static verifyPrevPeriodProof(proof) {
```

Comment:
the name is misleading IMO. It is not verifying anything, but rather calculating period root for the proof

Reply:
Yes this is because in solidity the verify method just computes the edges if the walk and verifies they reference one another. See new commit for clarification.